common/scummsys.h

```diff
@@ -480,57 +480,33 @@
 
 
 //
-// Typedef our system types unless they have already been defined by config.h,
-// or SCUMMVM_DONT_DEFINE_TYPES is set.
+// Typedef our system types unless SCUMMVM_DONT_DEFINE_TYPES is set.
+// This is usually due to conflicts with the system headers.
 //
-#if !defined(HAVE_CONFIG_H) && !defined(SCUMMVM_DONT_DEFINE_TYPES)
+#ifndef SCUMMVM_DONT_DEFINE_TYPES
 	typedef unsigned char byte;
-	typedef unsigned char uint8;
-	typedef signed char int8;
-	typedef unsigned short uint16;
-	typedef signed short int16;
-	// HACK: Some ports, such as NDS and AmigaOS, are not frequently
-	// tested during development, but cause frequent buildbot failures
-	// because they need to use 'long' for int32. Windows 32-bit
-	// binaries have this nice property of being easy to build, having
-	// a 32-bit 'long' too, *and* being frequently tested (incl. Github
-	// Actions). We want to catch this case as early and frequently
-	// as possible, so Win32 is probably the best candidate for this...
-	#if defined(WIN32) && !defined(_WIN64)
-	typedef unsigned long uint32;
-	typedef signed long int32;
-	#else
-	typedef unsigned int uint32;
-	typedef signed int int32;
-	#endif
 	typedef unsigned int uint;
-	typedef signed long long int64;
-	typedef unsigned long long uint64;
-#endif
-
-//
-// Determine 64 bitness
-// Reference: https://web.archive.org/web/20190413073704/http://nadeausoftware.com/articles/2012/02/c_c_tip_how_detect_processor_type_using_compiler_predefined_macros
-//
-#if !defined(HAVE_CONFIG_H)
-
-#if defined(__x86_64__) || \
-		  defined(_M_X64) || \
-		  defined(__ppc64__) || \
-		  defined(__powerpc64__) || \
-		  defined(__LP64__) || \
-		  defined(_M_ARM64)
-
-typedef int64 intptr;
-typedef uint64 uintptr;
 
+	typedef uint8_t uint8;
+	typedef int8_t int8;
+	typedef uint16_t uint16;
+	typedef int16_t int16;
+#if defined(__MORPHOS__) || defined(__amigaos4__)
```

Comment:
If I understand it correctly, this is what @dwatteau is proposing, in order to keep the existing type hack:

```suggestion
// HACK: Some ports, such as NDS and AmigaOS, are not frequently
// tested during development, but cause frequent buildbot failures
// because they need to use 'long' for int32. Windows 32-bit
// binaries have this nice property of being easy to build, having
// a 32-bit 'long' too, *and* being frequently tested (incl. Github
// Actions). We want to catch this case as early and frequently
// as possible, so Win32 is probably the best candidate for this...
#if (defined(WIN32) && !defined(_WIN64)) || defined(__MORPHOS__) || defined(__amigaos4__)
```

Reply:
Just talked with @ccawley2011, and this block is meant to be used as a temporary workaround, as stated in the original description

Reply:
Sadly, this cannot be removed and this is by no means a temporary workaround.

AmigaOS defines int32: `typedef   signed long  int32;`
But in the same time, newlib (their libc) defines int32_t: `typedef signed int int32_t;`
When we define int32, we must have a match with AmigaOS definition because we sometimes include system headers along our scummsys.h.

So, this PR can't fix the discrepancies problems we have on a regular basis.
To detect this before the buildbot, we need a platform which accepts `long int` as 32 bits sized int, this means a LLP64 platform and only Windows fits.
I am not sure we want to define it like this but maybe have a special define to enable it?

Reply:
Thanks for your analysis!

Yes the win32 "hack" could be enabled only with a special opt-in flag. I chose to hardcode it back then, because I thought it could improve our internal coverage; i.e. some devs use MSVC and so having real daily users hit the int/int32 case before any CI triggers.

Just to help catching the issue as early as possible, so that someone doesn't start working on a new engine for months, before hitting CI/buildbot issues much later. The sooner you catch it the easier it is to fix it.

Also, if it's integrated into MSVC as well, it means that it may become easier to fix the issue, by iterating with the IDE, instead of "blind" Github Actions usage.

But we might as well have a buildbot having a similar hack or something, yes :)